crates/rattler_index/src/lib.rs

```diff
@@ -443,22 +532,57 @@ pub async fn index<T: Configurator>(
         subdirs.insert(Platform::NoArch);
     }
 
+    let repodata_patch = if let Some(path) = repodata_patch {
+        match ArchiveType::try_from(path.clone()) {
+            Some(ArchiveType::Conda) => {}
+            Some(ArchiveType::TarBz2) | None => {
+                return Err(anyhow::anyhow!(
+                    "Only .conda packages are supported for repodata patches. Got: {}",
+                    path
+                ))
+            }
+        }
+        let repodata_patch_path = format!("noarch/{path}");
+        let repodata_patch_bytes = op.read(&repodata_patch_path).await?.to_bytes();
+        let reader = Cursor::new(repodata_patch_bytes);
+        let repodata_patch = repodata_patch_from_conda_package_stream(reader)?;
+        Some(repodata_patch)
+    } else {
+        None
+    };
+
     let semaphore = Semaphore::new(max_parallel);
     let semaphore = Arc::new(semaphore);
 
-    let tasks = subdirs
-        .iter()
-        .map(|subdir| {
-            tokio::spawn(index_subdir(
-                *subdir,
-                op.clone(),
-                force,
-                multi_progress.clone(),
-                semaphore.clone(),
-            ))
-        })
-        .collect::<Vec<_>>();
-    try_join_all(tasks).await?;
```

Comment:
this was silently failing before because it only checked whether the join handles were okay but not whether the actual return values of the tasks were